image_tools/src/cam2image.cpp

```diff
@@ -71,92 +67,80 @@ void convert_frame_to_message(
     rclcpp::QoSInitialization(
-      // The history policy determines how messages are saved until taken by the reader.
-      // KEEP_ALL saves all messages until they are taken, up to a system resource limit.
-      // KEEP_LAST enforces a limit on the number of messages that are saved.
-      // The limit is specified by the history "depth" parameter.
-      history_policy,
-      // Depth represents how many messages to save in history when the history policy is KEEP_LAST.
-      depth));
-
-  // The reliability policy can be reliable, meaning that the underlying transport layer will try
-  // ensure that every message gets received in order, or best effort, meaning that the transport
-  // makes no guarantees about the order or reliability of delivery.
-  qos.reliability(reliability_policy);
-
-  RCLCPP_INFO(node_logger, "Publishing data on topic '%s'", topic.c_str());
-  // Create the image publisher with our custom QoS profile.
-  auto pub = node->create_publisher<sensor_msgs::msg::Image>(topic, qos);
+      history_policy_,
+      depth_));
+  qos.reliability(reliability_policy_);
+  RCLCPP_INFO(node_logger, "Publishing data on topic '%s'", topic_.c_str());
+  pub_ = create_publisher<sensor_msgs::msg::Image>(topic_, qos);
 
   // is_flipped will cause the incoming camera image message to flip about the y-axis.
   bool is_flipped = false;
+  auto callback = [&is_flipped, &node_logger](const std_msgs::msg::Bool::SharedPtr msg) -> void
+  {
+    is_flipped = msg->data;
+    RCLCPP_INFO(node_logger, "Set flip mode to: %s", is_flipped ? "on" : "off");
+  };
 
-  // Subscribe to a message that will toggle flipping or not flipping, and manage the state in a
-  // callback.
-  auto callback =
-    [&is_flipped, &node_logger](const std_msgs::msg::Bool::SharedPtr msg) -> void
-    {
-      is_flipped = msg->data;
-      RCLCPP_INFO(node_logger, "Set flip mode to: %s", is_flipped ? "on" : "off");
-    };
-
-  // Set the QoS profile for the subscription to the flip message.
-  auto sub = node->create_subscription<std_msgs::msg::Bool>(
+  sub_ = create_subscription<std_msgs::msg::Bool>(
     "flip_image", rclcpp::SensorDataQoS(), callback);
 
   // Set a loop rate for our main event loop.
-  rclcpp::WallRate loop_rate(freq);
+  rclcpp::WallRate loop_rate(freq_);
 
   cv::VideoCapture cap;
   burger::Burger burger_cap;
-  if (!burger_mode) {
+  if (!burger_mode_) {
     // Initialize OpenCV video capture stream.
     // Always open device 0.
     cap.open(0);
 
     // Set the width and height based on command line arguments.
-    cap.set(cv::CAP_PROP_FRAME_WIDTH, static_cast<double>(width));
-    cap.set(cv::CAP_PROP_FRAME_HEIGHT, static_cast<double>(height));
+    cap.set(cv::CAP_PROP_FRAME_WIDTH, static_cast<double>(width_));
+    cap.set(cv::CAP_PROP_FRAME_HEIGHT, static_cast<double>(height_));
     if (!cap.isOpened()) {
       RCLCPP_ERROR(node_logger, "Could not open video stream");
-      return 1;
+      // return 1;
```

Comment:
It would be good if the demo stops running when there is an error (same as the previous behavior). I think we can achieve this by throwing an exception here:

```c++
throw std::runtime_error("Could not open video stream");
```

I would still keep the log macro since it looks like the component API does not echo the exception message.